main/java/com/bchupika/user/service/UserServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.bchupika.user.service;
+
+import com.bchupika.user.dto.UserRegistrationDto;
+import com.bchupika.user.dto.UserResponseDto;
+import com.bchupika.user.entity.User;
+import com.bchupika.user.repository.UserRepositoryImpl;
+import com.bchupika.user.validation.UserValidator;
+
+import java.util.Optional;
+
+public class UserServiceImpl implements UserService {
+    private final UserRepositoryImpl userRepository;
+    private final UserValidator userValidator;
+
+    public UserServiceImpl(UserRepositoryImpl userRepository, UserValidator userValidator) {
+        this.userRepository = userRepository;
+        this.userValidator = userValidator;
+    }
+
+    @Override
+    public UserResponseDto registerUser(UserRegistrationDto userRegistrationDto) {
+        this.userValidator.isValid(userRegistrationDto);
+        Long id = (Long) this.userRepository.save(userRegistrationDto);
+
+        return this.getUserById(id);
+    }
+
+    @Override
+    public UserResponseDto getUserById(Long userId) {
+        Optional<User> user = this.userRepository.getUserById(userId);
+        if (user.isEmpty()) {
+            return new UserResponseDto(0L, "", "", "");
+        }
```

Comment:
If user is not present in DB it is better to throw an exception, for example your custom `EntityNotFoundException`. Because such dummy data `new UserResponseDto(0L, "", "", "")` doesn't make any sense from the business logic perspective.

Reply:
This method could be implemented in such a way:
```
public UserResponseDto getUserResponseById(Long userId) {
    return this.userRepository.getUserById(userId)
            .map(user -> new UserResponseDto(user.getId(), user.getName(), user.getEmail(), user.getPhoneNumber()))
            .orElseThrow(() -> new EntityNotFoundException("User not found with ID: " + userId));
}
```